### ISSUE TITLE
Vertical regridding to reference levels fails when reference dataset needs to be downloaded

### ISSUE THREAD

**Issue by schlunma** [ESMValGroup/ESMValCore#1454]
**Describe the bug**

A recipe with the following preprocessor

```yml
  preprocessor_1:
    extract_levels:
      levels: bcc-csm1-1
      scheme: linear
```

fails when the reference dataset (here `bcc-csm1-1`) needs to be downloaded (with `offline=False`) since the loading of this dataset already happens in `_recipe.py` prior to downloading:

https://github.com/ESMValGroup/ESMValCore/blob/7d430722af8a3aee206287edb210eedc87ca4fcb/esmvalcore/_recipe.py#L164-L172

**Please attach**
[recipe_test_reference_levels.yml.txt](https://github.com/ESMValGroup/ESMValCore/files/7985530/recipe_test_reference_levels.yml.txt)
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/7985531/main_log_debug.txt)

**Comment by schlunma** [comment]
Moving this to v2.6 to follow the corresponding PR.

**Comment by remi-kazeroni** [comment]
Hi @schlunma, I understand this would be address by #1609 which is receiving a lot of attention at the moment. Have you tested that the issue reported here is fixed with #1609?

**Comment by schlunma** [comment]
Yes, see https://github.com/ESMValGroup/ESMValCore/pull/1609#issuecomment-1420711714
